x-pack/plugins/ml/public/application/services/toast_notification_service.ts

```diff
@@ -0,0 +1,84 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { ToastInput, ToastOptions, ToastsStart } from 'kibana/public';
+import { ResponseError } from 'kibana/server';
+import { useMemo } from 'react';
+import { useNotifications } from '../contexts/kibana';
+import {
+  BoomResponse,
+  extractErrorProperties,
+  MLCustomHttpResponseOptions,
+  MLErrorObject,
+  MLResponseError,
+} from '../../../common/util/errors';
+
+export type ToastNotificationService = ReturnType<typeof toastNotificationServiceProvider>;
+
+export function toastNotificationServiceProvider(toastNotifications: ToastsStart) {
+  return {
+    displaySuccessToast(toastOrTitle: ToastInput, options?: ToastOptions) {
+      toastNotifications.addSuccess(toastOrTitle, options);
+    },
+
+    displayErrorToast(error: any, toastTitle: string) {
+      const errorObj = this.parseErrorMessage(error);
+      if (errorObj.fullErrorMessage !== undefined) {
+        // Provide access to the full error message via the 'See full error' button.
+        toastNotifications.addError(new Error(errorObj.fullErrorMessage), {
+          title: toastTitle,
+          toastMessage: errorObj.message,
+        });
+      } else {
+        toastNotifications.addDanger(
+          {
+            title: toastTitle,
+            text: errorObj.message,
+          },
+          { toastLifeTimeMs: 30000 }
+        );
+      }
+    },
+
+    parseErrorMessage(
+      error:
+        | MLCustomHttpResponseOptions<MLResponseError | ResponseError | BoomResponse>
+        | undefined
+        | string
+        | MLResponseError
+    ): MLErrorObject {
+      if (
+        typeof error === 'object' &&
+        'response' in error &&
+        typeof error.response === 'string' &&
+        'statusCode' in error
```

Comment:
using `in` here will be `true` even if it is `undefined`
This should be a check for `undefined` if we need `statusCode` to exist

Reply:
Done in https://github.com/elastic/kibana/pull/71618/commits/5ba2d779b365092d9a3b3d381df989d55d39bd96